### ISSUE TITLE
Design the relational database

### ISSUE THREAD

**Issue by jessieAnhNguyen** [jessieAnhNguyen/Grocery-Cache#2]
- Create a table for the general grocery list 
- Create table for each category and list of items
- Connect the two relations by normalization
- Draw the ER diagram

**Comment by jessieAnhNguyen** [comment]
## Database Design
### Three tables in the database
1. Main_list: stores the basic information for each item (item_name, category, budget, urgency_level (1-5), notes). Primary key: item_id, Foreign key: category_id
2. Category: stores the information about each category (category_name, description). Primary key: category_id
3. Category_list: Join relationship between 2 tables. 

### Interfaces
- In the main list, display all the items and their information (one item in each role). Display the category_name instead of the category_id for readability.
- In each specific category list, let the user choose a specific category they want to view. After that, select all the items in each category (one item in each role). Display the description of the category on top of the list.

**Comment by jessieAnhNguyen** [comment]
### ER Relation Diagram
![image](https://user-images.githubusercontent.com/47696418/99151143-bada0900-26cb-11eb-94b6-41d41edf827e.png)

**Comment by jessieAnhNguyen** [comment]
@amnshrestha @anindini Please let me know if this looks good! Especially @amnshrestha if this is enough for your first template

**Comment by anindini** [comment]
This looks good! Will you also be adding a "specific recipes" list? Also, what is "Notes" for?

**Comment by jessieAnhNguyen** [comment]
Thanks! So the "specific recipes" would be in "specific category list" above. The idea is we let the user create a category list, and then later they can choose to view which category. This will show all items in that category. We can discuss more in our next meeting!

"Notes" is a text field that user can add anything they want (for example: for fertilizer, they can note that "buy 3 packs" or sth)

**Comment by shrestha-aman** [comment]
1.  Is "Category" in Main_List the same as "Category_name" in Category?
2. Also, one item can be in multiple categories. I don't know if Category_list is taking care of that. 
     We might need an additional table for item and category if Category_list is not taking care of it. 
3. There's a dependency between Item_Id and Item_Name. Do we want to keep this in another table?
4. The relation of Category_List. Is it many to many on both sides? (Wanted to make sure) If so, we might need to keep a Join Table

I'm gonna work on a general template based on what I see here but we can make changes to the template as per the changes we make to this diagram.

**Comment by jessieAnhNguyen** [comment]
1. Yes.
2. Yes, Category_list is taking care of one item in multiple categories. So my idea for Category_list is: full join table for Main_list and Category. When the user chooses a specific category list to display (for ex, "momo"), then we SELECT all items from Category_list table that has a category of "momo".
3. I'm not sure what you mean, can you elaborate please?
4. Yes, Category_List is it many to many on both sides, and we should keep a Join table (as my answer to 2)

Let me know if this makes sense and we can discuss further!

**Comment by jessieAnhNguyen** [comment]
As we have discussed, we came to this conclusion of the database design:
- When user adds an item, they add all information about it. They can choose the category from a drop-down list. 

- If it's not there, they can create a new category. We will navigate user to another page to add a new category. We also add this entry into the **Category** table.

- We will add this entry (without category) into the **Item_list** table. We will add an entry (Item_ID, Category_ID) into the **Category_list** table.

![image](https://user-images.githubusercontent.com/47696418/99618890-ddb24780-2a54-11eb-8ab6-e0e93bd68c78.png)
